src/DIRAC/WorkloadManagementSystem/Client/JobState/CachedJobState.py

```diff
@@ -116,12 +116,15 @@ def serialize(self):
       manifest = [self.__manifest.dumpAsCFG(), self.__manifest.isDirty()]
     else:
       manifest = None
-    return DEncode.encode((self.__jid, self.__cache, self.__jobLog, manifest,
-                           self.__initState, self.__insertIntoTQ, list(self.__dirtyKeys)))
+    data = DEncode.encode((
+        self.__jid, self.__cache, self.__jobLog, manifest, self.__initState,
+        self.__insertIntoTQ, list(self.__dirtyKeys)
+    ))
+    return data.decode()
```

Comment:
What are these fields likely to be? I guess it will never be arbitrary binary data? Should there be a issue to make this use JSON instead of DEncode?

Reply:
It's probably possible, but not for today...